File: services/api_service/src/main/java/io/lvoxx/ssurl/api_service/service/AuthService.java
package io.lvoxx.ssurl.api_service.service;

import io.lvoxx.ssurl.common.dto.request.LoginRequest;
import io.lvoxx.ssurl.common.dto.request.RegisterRequest;
import io.lvoxx.ssurl.common.dto.response.AuthResponse;
import io.lvoxx.ssurl.common.dto.response.UserResponse;
import org.springframework.http.server.reactive.ServerHttpResponse;
import reactor.core.publisher.Mono;

public interface AuthService {

    Mono<UserResponse> register(RegisterRequest request);

    Mono<AuthResponse> login(LoginRequest request, ServerHttpResponse response);

    Mono<AuthResponse> refresh(String refreshToken);

    Mono<Void> logout(String refreshToken, ServerHttpResponse response);
}

File: shorten-url-app/shared/services/AuthService.ts
import { ApiService } from './ApiService'
import type { AuthResponse, LoginRequest, RegisterRequest, User } from '../types'

export class AuthService {
  constructor(private api: ApiService) {}

  async register(data: RegisterRequest): Promise<User> {
    return this.api.post<User>('/api/v1/auth/register', data)
  }

  async login(data: LoginRequest): Promise<AuthResponse> {
    return this.api.post<AuthResponse>('/api/v1/auth/login', data)
  }

  async refresh(): Promise<AuthResponse> {
    return this.api.post<AuthResponse>('/api/v1/auth/refresh')
  }

  async logout(): Promise<void> {
    await this.api.post('/api/v1/auth/logout')
  }
}
